packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Component/Repeatable.tsx

```diff
@@ -211,6 +207,8 @@ const RepeatableComponent = ({
       <AccordionGroup error={error}>
         <AccordionContent aria-describedby={ariaDescriptionId}>
           {value.map(({ __temp_key__: key, id }, index) => {
+            const nameWithIndex = `${name}.${index}`;
+            const safeKey = key ?? nameWithIndex;
```

Comment:
I'd like to avoid creating keys like this if possible, we had this in V4 and it got very messy and quite buggy to follow, especially if it leaks into the EditView, the re-ordering will crash.

We create tmp keys when we fetch the document: 

https://github.com/strapi/strapi/blob/4300e8f16c86f11fe187f56b72eabee2f5d739b9/packages/core/content-manager/admin/src/pages/EditView/EditViewPage.tsx#L122

https://github.com/strapi/strapi/blob/4300e8f16c86f11fe187f56b72eabee2f5d739b9/packages/core/content-manager/admin/src/pages/EditView/utils/data.ts#L127

You could use the same pipe funcs perhaps? or maybe just the tmp key one.